src/Ztm.Hosting/BackgroundServiceErrorLogger.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Ztm.Hosting
+{
+    public sealed class BackgroundServiceErrorLogger : IBackgroundServiceExceptionHandler
+    {
+        readonly ILogger logger;
+
+        public BackgroundServiceErrorLogger(ILogger<BackgroundServiceErrorLogger> logger)
+        {
+            if (logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            this.logger = logger;
+        }
+
+        public Task RunAsync(Type service, Exception exception, CancellationToken cancellationToken)
+        {
+            if (service == null)
+            {
+                throw new ArgumentNullException(nameof(service));
+            }
+
+            if (exception == null)
+            {
+                throw new ArgumentNullException(nameof(exception));
+            }
```

Comment:
I think we could make handler to be abstract class to let us make two checkings above be shared.